R/slice.R

```diff
@@ -4,8 +4,11 @@
 #' for all vector types, regardless of dimensionality. It is an analog to `[`
 #' that matches [vec_size()] instead of `length()`.
```

Comment:
Maybe worth adding a new section that lists all differences with `[`? i.e. only slices along one dimension, preserves attributes of unclassed types, ...